src/utils/gitUtils.ts

```diff
@@ -21,10 +21,25 @@ import { getSingleRootFsPath } from './workspaceUtils';
 export type GitWorkspaceState = { repo: Repository | null, dirty: boolean, remoteRepo: ReposGetResponseData | undefined; hasAdminAccess: boolean };
 export type VerifiedGitWorkspaceState = GitWorkspaceState & { repo: Repository };
 
+// use only when using the gitApi
+export async function callWithGitErrorHandling<T>(callback: () => Promise<T>): Promise<T | null> {
```

Comment:
Why we usin `null` instead of `undefined`?

Reply:
All the gitApi command (specifically git.init and git.GetRepository) return `Repository | null` so just had to do it for typing reasons.